exporter.go

```diff
@@ -672,12 +676,12 @@ func (e *Exporter) addOrReuseExtensionLayer(image imgutil.Image, layer layers.La
 		}
 		e.Logger.Infof("Adding extension layer %s\n", layer.ID)
 		e.Logger.Debugf("Layer '%s' SHA: %s\n", layer.ID, layer.Digest)
-		return layer.Digest, image.AddLayerWithDiffID(layer.TarPath, layer.Digest)
+		return layer.Digest, image.AddLayerWithDiffIDAndHistory(layer.TarPath, layer.Digest, layer.History)
 	}
 	_ = rc.Close() // close the layer reader
 	e.Logger.Infof("Reusing layer %s\n", layer.ID)
 	e.Logger.Debugf("Layer '%s' SHA: %s\n", layer.ID, layer.Digest)
-	return layer.Digest, image.ReuseLayer(layer.Digest)
+	return layer.Digest, image.ReuseLayer(layer.Digest) // TODO: ReuseLayerWithHistory
```

Comment:
We need the ability to add history when re-using a layer so that re-builds where the previous image was built on an earlier platform will still have history in the re-used layers